docs/diagrams/catchup-procedure.puml

```diff
@@ -0,0 +1,317 @@
+@startuml
+!pragma teoz true
+hide footbox
+
+participant Node1
+participant Node2
+participant Node3
+participant Node4
+
+note over Node1, Node4
+    <b>Catch-up of Node4
+end note
+
+Node4 -> Node4: Revert Unordered 3PC Batches
+note right Node4
+    <b>Revert</b> applied but not yet ordered
+    3PC batches on Master replica
+end note
+
+Node4 -> Node4: mode = starting
+note right Node4
+    While  <b>mode != participating:
+        - Replicas will discard incoming PRE-PREPAREs,
+        - Node will stash ORDEREDs coming from Replicas
+end note
+
+loop For each ledger
+
+    note over Node1, Node4
+        <b>Catch-up of a particular ledger
+    end note
+
+    Node4 ->> Node1: MESSAGE_REQUEST(LEDGER_STATUS)
+    Node4 ->> Node2: MESSAGE_REQUEST(LEDGER_STATUS)
+    Node4 ->> Node3: MESSAGE_REQUEST(LEDGER_STATUS)
+    note right Node4
+        Ask other nodes for  <b>their LEDGER_STATUSes</b>
+    end note
+
+    Node1 ->> Node4: MESSAGE_RESPONSE(LEDGER_STATUS)
+    Node2 ->> Node4: MESSAGE_RESPONSE(LEDGER_STATUS)
+    Node3 ->> Node4: MESSAGE_RESPONSE(LEDGER_STATUS)
+    note right Node4
+        <b>LEDGER_STATUS contains:
+            - <b>txn_seq_no</b> <i>(seq_no of last transaction in the ledger),
+            - <b>(view_no, pre_prepare_seq_no)</b>
+              <i>(key of Master's 3PC batch which contained last transaction),
+            - <b>merkle root</b> <i>(i.e. ledger root)
+    end note
+
+    alt At least n-f-1 nodes replied with not newer LEDGER_STATUSes
+
+        note over Node1, Node4
+            Node finds its ledger up to date
+        end note
+
+        Node4 -> Node4: Process stashed ORDEREDs
```

Comment:
Please describe what is stashed ordered, why we need to process them, and how they may appear here.